ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/StartupUtil.java

```diff
@@ -52,9 +51,12 @@ public static BeaconState createMockedStartInitialBeaconState(
         .createInitialBeaconState(UnsignedLong.valueOf(genesisTime), initialDepositData);
   }
 
-  public static BeaconState loadBeaconStateFromFile(final String stateFile) throws IOException {
+  private static BeaconState loadBeaconState(final String source) throws IOException {
     return SimpleOffsetSerializer.deserialize(
-        Bytes.wrap(Files.readAllBytes(new File(stateFile).toPath())), BeaconStateImpl.class);
+        ResourceLoader.urlOrFile()
+            .loadBytes(source)
+            .orElseThrow(() -> new FileNotFoundException("Could not find " + source)),
```

Comment:
Is there a better exception, it seems like the source is not necessarily a File.

Reply:
We could just throw `IllegalArgumentException` but the final fallback is to treat it as a file so this matches the previous behaviour and winds up being fairly intuitive for users.